components/external_storage/src/lib.rs

```diff
@@ -44,25 +45,27 @@ pub fn record_storage_create(start: Instant, storage: &dyn ExternalStorage) {
         .observe(start.saturating_elapsed().as_secs_f64());
 }
 
+/// UnpinReader is a simple wrapper for AsyncRead + Unpin + Send.
+/// This wrapper would remove the lifetime at the argument of the generted async function
+/// in order to make rustc happy. (And reduce the length of signture of write.)
+/// see https://github.com/rust-lang/rust/issues/63033
+pub struct UnpinReader(pub Box<dyn AsyncRead + Unpin + Send>);
```

Comment:
```suggestion
pub type UnpinReader = Box<dyn AsyncRead + Unpin + Send + 'static>;
```

?

Reply:
I'm afraid that we must use the new type pattern but type alias for resolving the lifetime deriving problem... Even through let me try it...🤔